### ISSUE TITLE
git cola lags when I select a different repository in the favorites list

### ISSUE THREAD

**Issue by djhaskin987** [git-cola/git-cola#498]
I don't know if this is a big issue, but I wanted to bring it to your attention @davvid . Thanks so much for this great tool.

I have a build of git cola before the icons were changed, and it seems this build does not have the problem.

After the icons were changed, When I click on one repository in my favorites list, then click on a different one, or when I open one repo up and then select a different repo from my favorites list, git cola locks up, and becomes unresponsive sometimes. Or, when I select a different repo and then select another repository in rapid succession from my favorites list, the browser widget displays all the files as "Modified", sometimes for whole minutes after I have switched to a different repository in my favorites list. I don't know if the two phenomena are related (Modified problem and lagging problem), but the lagging is my main concern here. 

So, to summarize:

1) Open up one repo using the startup widget from the favorites list. It comes right up, no problem.
2) Open up a different one using the favorites widget. It may or may not hang when you do so. Perhaps this hanging depends on the size of the repository. I have a large favorites list with many repositories in it.

**Comment by djhaskin987** [comment]
Also: When I start up git cola, sometimes it uses 123% CPU and locks up. Again, I noticed this after the icons changed, in the past week or so.

**Comment by djhaskin987** [comment]
K, I have more information on the bug.

I have narrowed it down. Whether I select a repository from the favorites list, or open it via the startup widget, this _one_ repository I have makes git-cola take up 124% CPU. After removing a virtualenv directory, a debian package, some other files, and a *.orig file from the repository, I can open it right up again in git cola, no problems.

Ideas?

**Comment by davvid** [comment]
For that one repository, it sounds like most of the time is spend looking at untracked files.  There is an option to skip showing untracked files in the preferences.

It would also be good to rework it so that untracked files are gathered as a background task, rather than blocking for it.  That would probably make a huge difference.

**Comment by djhaskin987** [comment]
My coworker also noticed this. We find it only happens in a particularly gnarly python project, and only then because some of the generated files mess with PYTHON_PATH . This causes git cola to flip out. I'll see if I can reproduce with a simpler example.

**Comment by almoore** [comment]
There is an issue where when there are a lot of files in the repo that are not tracked or ignored, each of files is checked or something and all theses file checks take a long time. The command line git status will just display the file path if is not tracked and therefor take a lot less time because it does not run checks on them.

**Comment by djhaskin987** [comment]
Yes precisely. The issue I've run into is the one described by @almoore . Sorry it was hard to describe.

**Comment by davvid** [comment]
Indeed, `.gitignore` can make a big difference. You can make things faster in that repo by going into it and running `git config gui.displayuntracked false` and it'll disable building all of the widgets and performing the checks on the untracked files.

Please test that out. I'm going to close this issue as a duplicate of #499 so that we can gather the notes in one place.
